components/applications-service/pkg/nats/nats.go

```diff
@@ -130,7 +131,10 @@ func (nc *NatsClient) Connect() error {
 			"max_retries": nc.retries,
 		}).Error("Unable to connect to server")
 
-		time.Sleep(5 * time.Second)
+		baseSleep := int64(uint32(1) << tries)
+		randomization := rand.Int63n(baseSleep)
+		totalSleep := baseSleep + randomization
```

Comment:
Randomized exponential backoff. This breaks up simultaneous requests over time as it keeps retrying.